notebooks/tutorial_ivf_pq.ipynb

```diff
@@ -0,0 +1,1226 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# RAFT's DeviceResources controls the GPU, cuda stream, memory policies etc.\n",
+    "# For now, we just create a default instance.\n",
+    "from pylibraft.common import DeviceResources\n",
+    "resources = DeviceResources()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'add_data_on_build': True,\n",
+       " 'codebook_kind': 0,\n",
+       " 'conservative_memory_allocation': False,\n",
+       " 'force_random_rotation': False,\n",
+       " 'kmeans_n_iters': 20,\n",
+       " 'kmeans_trainset_fraction': 0.5,\n",
+       " 'metric': 1,\n",
+       " 'n_lists': 1024,\n",
+       " 'pq_bits': 8,\n",
+       " 'pq_dim': 64}"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from pylibraft.neighbors import ivf_pq\n",
+    "# First, we need to initialize the build/indexing parameters.\n",
+    "# One of the more important parameters is the product quantisation (PQ) dim.\n",
+    "# Effectively, this parameter says\n",
+    "#      \"shrink the dataset to this dimensionality to reduce the index size\".\n",
+    "# It must be not bigger than the dataset dim,\n",
+    "# and it should be divisible by 32 for better GPU performance.\n",
+    "pq_dim = 1\n",
+    "while pq_dim * 2 < dataset.shape[1]:\n",
+    "    pq_dim = pq_dim * 2\n",
+    "# We'll use the ANN-benchmarks-provided metric and sensible defaults for the rest of parameters.\n",
+    "index_params = ivf_pq.IndexParams(n_lists=1024, metric=metric, pq_dim=pq_dim)\n",
+    "\n",
+    "show_properties(index_params)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 1.32 s, sys: 42.7 ms, total: 1.36 s\n",
+      "Wall time: 1.35 s\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Index(type=IVF-PQ, metric=euclidean, codebook=subspace, size=1000000, dim=128, pq_dim=64, pq_bits=8, n_lists=1024, rot_dim=128)"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "## Build the index\n",
+    "# This function takes a row-major either numpy or cupy (GPU) array.\n",
+    "# Generally, it's a bit faster with GPU inputs, but the CPU version may come in handy\n",
+    "# if the whole dataset cannot fit into GPU memory (or even CPU RAM -- use mmap for that).\n",
+    "index = ivf_pq.build(index_params, dataset, handle=resources)\n",
+    "# Sync the GPU to make sure we've got the timing right\n",
+    "resources.sync()\n",
+    "index"
+   ]
+  },
```

Comment:
After the index is built, please demonstrate how to save and load the index (reference: https://github.com/rapidsai/raft/blob/0abedc6661748bd64ff9828a86304dd29ae0ca05/python/pylibraft/pylibraft/test/test_ivf_pq.py#L518-L520)